tests/qg_test_utils.py

```diff
@@ -0,0 +1,213 @@
+    initOutput = cT.InitOutput(name="Dummy1InitOutput", storageClass="ExposureF", doc="n/a")
+    input = cT.Input(name="Dummy1Input", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+    output = cT.Output(name="Dummy1Output", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+
+
+class Dummy1Config(PipelineTaskConfig, pipelineConnections=Dummy1Connections):
+    conf1 = Field(dtype=int, default=1, doc="dummy config")
+
+
+class Dummy1PipelineTask(PipelineTask):
+    ConfigClass = Dummy1Config
+
+
+class Dummy2Connections(PipelineTaskConnections, dimensions=("D1", "D2")):
+    initInput = cT.InitInput(name="Dummy1InitOutput", storageClass="ExposureF", doc="n/a")
+    initOutput = cT.InitOutput(name="Dummy2InitOutput", storageClass="ExposureF", doc="n/a")
+    input = cT.Input(name="Dummy1Output", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+    output = cT.Output(name="Dummy2Output", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+
+
+class Dummy2Config(PipelineTaskConfig, pipelineConnections=Dummy2Connections):
+    conf1 = Field(dtype=int, default=1, doc="dummy config")
+
+
+class Dummy2PipelineTask(PipelineTask):
+    ConfigClass = Dummy2Config
+
+
+class Dummy3Connections(PipelineTaskConnections, dimensions=("D1", "D2")):
+    initInput = cT.InitInput(name="Dummy2InitOutput", storageClass="ExposureF", doc="n/a")
+    initOutput = cT.InitOutput(name="Dummy3InitOutput", storageClass="ExposureF", doc="n/a")
+    input = cT.Input(name="Dummy2Output", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+    output = cT.Output(name="Dummy3Output", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+
+
+class Dummy3Config(PipelineTaskConfig, pipelineConnections=Dummy3Connections):
+    conf1 = Field(dtype=int, default=1, doc="dummy config")
+
+
+class Dummy3PipelineTask(PipelineTask):
+    ConfigClass = Dummy3Config
+
+
+# Test if a Task that does not interact with the other Tasks works fine in
+# the graph.
+class Dummy4Connections(PipelineTaskConnections, dimensions=("D1", "D2")):
+    input = cT.Input(name="Dummy4Input", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+    output = cT.Output(name="Dummy4Output", storageClass="ExposureF", doc="n/a", dimensions=("D1", "D2"))
+
+
+class Dummy4Config(PipelineTaskConfig, pipelineConnections=Dummy4Connections):
+    conf1 = Field(dtype=int, default=1, doc="dummy config")
+
+
+class Dummy4PipelineTask(PipelineTask):
+    ConfigClass = Dummy4Config
+
+
+def make_test_quantum_graph():
+    """Create a QuantumGraph for unit tests.
+
+    Returns
+    -------
+    qgraph : `lsst.pipe.base.QuantumGraph`
+        A test QuantumGraph looking like the following:
+        (DummyTask4 is completely independent.)
+
+        T1(1,2)   T1(3,4)  T4(1,2)  T4(3,4)
+           |         |
+        T2(1,2)   T2(3,4)
+           |         |
+        T3(1,2)   T3(3,4)
+    """
```

Comment:
It's not clear to me what numbers in the parentheses mean. Could you add a brief explanation to the comment?